pkg/prometheus/promcfg.go

```diff
@@ -2134,6 +2151,10 @@ func (cg *ConfigGenerator) generateK8SSDConfig(
 		})
 	}
 
+	if ptr.Deref(selectorMechanism, monitoringv1.SelectorMechanismRelabel) == monitoringv1.SelectorMechanismRole {
+		k8sSDConfig = generateRoleSelectorConfig(k8sSDConfig, selectorRoles, selector, cg)
+	}
```

Comment:
(nit) we could always call generateRoleSelectorConfig() and have the test on the selector mechanism there.
```suggestion
	k8sSDConfig = generateRoleSelectorConfig(k8sSDConfig, selectorRoles, selector, cg)
```